tests/SCsub

```diff
@@ -13,9 +15,14 @@ env_tests.Append(CPPDEFINES=["DOCTEST_CONFIG_IMPLEMENT"])
 # Since we link with /MT thread_local is always expired when the header is used
 # So the debugger crashes the engine and it causes weird errors
 # Explained in https://github.com/onqtam/doctest/issues/401
-if env_tests["platform"] == "windows":
+if env["platform"] == "windows":
     env_tests.Append(CPPDEFINES=[("DOCTEST_THREAD_LOCAL", "")])
 
+if env["platform"] == "linuxbsd" and methods.using_gcc(env):
+    # Supressing "-Wsubobject-linkage" is needed for extending
+    # `doctest::ConsoleReporter` which resides in an anonymous namespace.
+    env_tests.Append(CCFLAGS=["-Wno-subobject-linkage"])
+
```

Comment:
I'm not sure how to fix this properly, seems to be specific to GCC according to my researches on this warning...

Perhaps using `methods.using_gcc(env)` check alone is fine too, made it specific to `linuxbsd` platform because that's the only platform where CI failed.

Reply:
I'd remove the platform check as it's likely to affect MinGW-gcc on Windows or macOS too.